staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -162,7 +174,10 @@ type Store struct {
 	DeleteStrategy rest.RESTDeleteStrategy
 	// AfterDelete implements a further operation to run after a resource is
 	// deleted and before it is decorated, optional.
-	AfterDelete ObjectFunc
+	//FIXME: pass whe whole options instead?
+	//FIXME: Do the same for other After* hooks?
+	//FIXME: if so, get rid of ObjectFunc type?
+	AfterDelete func(obj runtime.Object, dryRun bool)
```

Comment:
it seems a little weird that only this After func needs to be aware of dryRun.  Is this becasue there isn't a BeginDelete?

Reply:
Yes - because Delete is a split operation.  The Delete() method can complete, but the object isn't gone until either the grace period expires or some update removes a finalizer.

I'd be fine to plumb dryRun to other After* - it seems correct, but I wanted feedback.

Reply:
small bump here - you wanted the whole options passed to After*, which I agreed with.  The trick is that the split-delete means I don't have a DeleteOptions in all cases.  I can synthesize one from an UpdateOptions - fair?

Reply:
In the next push